### PR TITLE
Cleanup DEVMODE interop

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Change to unicode
- Cleanup definition (use enums etc)

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2026)

### CODE REVIEW

File: src/Common/src/Interop/User32/Interop.DEVMODEW.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using System.Windows.Forms;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
+        public struct DEVMODEW
+        {
+            private const int CCHDEVICENAME = 32;
```
Comment (zsd4yr) [inline src/Common/src/Interop/User32/Interop.DEVMODEW.cs]: https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-devmodea would you mind looking over this? it seems a little different

Comment (weltkante) [reply src/Common/src/Interop/User32/Interop.DEVMODEW.cs]: you are linking the ANSI version not the UNICODE version, [this](https://docs.microsoft.com/en-us/windows/win32/api/wingdi/ns-wingdi-devmodew) is the right link

The differences appear to be from inlining the unions, I looked over them and they seem ok

File: src/Common/src/Interop/User32/Interop.ENUM.cs [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+internal static partial class Interop
+{
+    internal static partial class User32
+    {
+        public enum ENUM : uint
```
Comment (RussKie) [inline src/Common/src/Interop/User32/Interop.ENUM.cs]: This naming... ummm... 🤢

Comment (hughbe) [reply src/Common/src/Interop/User32/Interop.ENUM.cs]: Yeah - horrible, but in line with our naming conventions... :(

**Comment by hughbe** [comment]
Is there any feedback needed here? Does not seem to be apparent

**Comment by RussKie** [comment]
@msftbot merge if @zsd4yr approves

File: src/Common/src/Interop/User32/Interop.DEVMODEW.cs (hunk 1 above)
Comment (zsd4yr) [reply src/Common/src/Interop/User32/Interop.DEVMODEW.cs]: ignorant question: should we inline the unions to match?

Comment (weltkante) [reply src/Common/src/Interop/User32/Interop.DEVMODEW.cs]: @zsd4yr I don't quite get what you are asking, the unions are inlined already and as far as I can tell they do match.

When inlining a union you pick one of the members and drop all the others. If you didn't pick the largest member you also may have to add some dummy padding member to get the right alignments, but as far as I can tell its not needed here.

The problem with unions is that C# doesn't have a great mapping for them. Sometimes you can do explicit struct layout with field offsets, but that only works if all members are primitives. For anything more complex you have to manually unpack the memory or work with pointers, both are not great.

In this specific case the union switches between monitor and printer, we inlined the monitor version so we are sacrificing support for reading the printer values. I assume WinForms either doesn't care about `DEVMODE` for printers or it has a separate copy of the struct for that.

Comment (zsd4yr) [reply src/Common/src/Interop/User32/Interop.DEVMODEW.cs]: Thanks @weltkante!

**Pull request merged**